core-4/src/main/java/parkinglot/AppConfig.java

```diff
@@ -21,11 +21,12 @@
 
 @Configuration
 public class AppConfig {
+    @Autowired
+    Map<String,Car> carMap;
```

Comment:
okay, looks much better, but we can do it the best way

now you place it as a variable, it is available for other beans, let's find a way to put it in the parking lot bean signature